### PR TITLE
General API docs improvements

### PR DESCRIPTION
**Pull request by robsdedude**
Including
 * more details
 * better examples
 * code formatting
 * more consistent docstring format

### CODE REVIEW

File: neo4j/_async/work/result.py [hunk 1]
```diff
@@ -521,60 +536,67 @@ async def graph(self) -> Graph:
...
+    async def data(self, *keys: _T_ResultKey) -> t.List[t.Dict[str, t.Any]]:
+        """Return the remainder of the result as a list of dictionaries.
+
+        This function provides a convenient but opinionated way to obtain the
+        remainder of the result as mostly JSON serializable data. It is mainly
+        useful for interactive sessions and rapid prototyping.
 
-        See :class:`neo4j.Record.data`
+        For instance, node and relationship labels are not included. You will
+        have to implement a custom serialzer should you need more control over
```
Comment (stefano-ottolenghi) [inline neo4j/_async/work/result.py]: typo in `serializer`

**Review by stefano-ottolenghi — CHANGES_REQUESTED** [review]

**Comment by stefano-ottolenghi** [comment]
Other than the typo, LGTM! 🦆

File: neo4j/_async/work/result.py (hunk 1 above)
Comment (robsdedude) [reply neo4j/_async/work/result.py]: fixed

**Review by stefano-ottolenghi — APPROVED** [review]

**Review by AndyHeap-NeoTech — APPROVED** [review]
All looks sensible and good to me. Approved.

**Pull request merged**